src/app/component/user/modal-auth/windowsignin/windowsignin.component.ts

```diff
@@ -0,0 +1,134 @@
+  loadingAnim: boolean;
+
+  emailErrorMessageBackEnd: string;
+  passwordErrorMessageBackEnd: string;
+
+  backEndError: string;
+
+  constructor(
+  private matDialogRef: MatDialogRef<WindowsigninComponent>,
+  private userOwnSignInService: UserOwnSignInService,
+  private router: Router,
+  private authService: AuthService,
+  private googleService: GoogleSignInService,
+  private localStorageService: LocalStorageService,
+  public dialog: MatDialog,
+  @Inject(MAT_DIALOG_DATA) public data: any
+
+  ) { }
+
+    readonly formPic = 'assets/img/formpic.jpg';
+    readonly picGoogle = 'assets/img/icon/google.svg';
+    readonly picArrow = 'assets/img/icon/arrows/google-arrow.svg';
+
+  ngOnInit() {
+  	this.userOwnSignIn = new UserOwnSignIn();
+    this.loadingAnim = false;
+    this.emailErrorMessageBackEnd = null;
+    this.passwordErrorMessageBackEnd = null;
+    this.backEndError = null;
+    this.localStorageService.userIdBehaviourSubject.subscribe(userId => {
+      if (userId) {
+        this.matDialogRef.close();
+      }
+    });
+  }
+
+   signIn(userOwnSignIn: UserOwnSignIn) {
+    this.loadingAnim = true;
+    this.userOwnSignInService.signIn(userOwnSignIn).subscribe(
+      (data: UserSuccessSignIn) => {
+        this.loadingAnim = false;
+        this.userOwnSignInService.saveUserToLocalStorage(data);
+        this.localStorageService.setFirstName(data.firstName);
+        this.localStorageService.setFirstSignIn();
+        this.router.navigate(['/welcome'])
+          .then(success => console.log('redirect has succeeded ' + success))
+          .catch(fail => console.log('redirect has failed ' + fail));
+      },
+      (errors: HttpErrorResponse) => {
+        try {
+          errors.error.forEach(error => {
+            if (error.name === 'email') {
+              this.emailErrorMessageBackEnd = error.message;
+            } else if (error.name === 'password') {
+              this.passwordErrorMessageBackEnd = error.message;
+            }
+          });
+        } catch (e) {
+          this.backEndError = errors.error.message;
+        }
+        this.loadingAnim = false;
+      }
+    );
+  }
+
+    signInWithGoogle() {
+    this.authService.signIn(GoogleLoginProvider.PROVIDER_ID).then(data => {
+      this.googleService.signIn(data.idToken).subscribe(
+        (data1: UserSuccessSignIn) => {
+          this.userOwnSignInService.saveUserToLocalStorage(data1);
+          this.router.navigate(['/']);
+        },
+        (errors: HttpErrorResponse) => {
+          try {
+            errors.error.forEach(error => {
+              if (error.name === 'email') {
+                this.emailErrorMessageBackEnd = error.message;
+              } else if (error.name === 'password') {
+                this.passwordErrorMessageBackEnd = error.message;
+              }
+            });
+          } catch (e) {
+            this.backEndError = errors.error.message;
+          }
+        }
+      );
+    });
+  }
+
+    clickImgSignIn() {
```

Comment:
It's not clear, please rename it

Reply:
how for example?

Reply:
this method should change password visibility, so it can be renamed like togglePasswordVisibility

Reply:
ready